**Case Description**

An FMCG brand X is available at a large number of stores.
The brand is avaialable as a basic version X and in the form of some value added variants
The price of brand X can vary from store to store depending on the dicounts/promos that the store is offering.
In some stores there are special displays for brand X

**Variable Desciprtion**

Price: Avg price of brand X in that store

Feature: Whether the value added variant of brand X is present in store(1) or not(0)

Display: Whether any store level promo of brand X is present in the store (1) or not(0)

Price-Competitors: Prices of 5 competition brands in the store

**Aim**

The aim of this project is to assess the impact the 4 Ps of marketing has on sales.

Here feature(indicates if the product is value added or not) is a proxy for product

display is a proxy for promotion

**Requirements & Metrics to Evaluate The Model**

Adj r^2>0.6

Mean Absolute Error

**Reading the dataframe**

In [2]:

import pandas as pd

# Read the Excel file
df_train = pd.read_excel('/content/Marketing_Mix Modelling Dataset.xlsx', sheet_name='MM Training data')
df_test = pd.read_excel('/content/Marketing_Mix Modelling Dataset.xlsx', sheet_name='MM Test data')

# Now you have two dataframes: df_train and df_test
# You can proceed with your analysis using these dataframes.


**Performing Step wise regression- Price V/S Sales & Evaluating Model Performance With MAE**

In [ ]:
import statsmodels.formula.api as sm

# Fit the linear regression model
model = sm.ols('Sales ~ Price', data=df_train).fit()

# Print the summary of the model
print(model.summary())

# Extract the coefficients
intercept = model.params['Intercept']
price_coeff = model.params['Price']

# Create the 'Predicted Sales' column in df_test
df_test['Predicted Sales'] = intercept + df_test['Price'] * price_coeff

# Calculate the Mean Absolute Error (MAE)
mae = (df_test['Sales'] - df_test['Predicted Sales']).abs().mean()

print(f"Mean Absolute Error (MAE): {mae}")




                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.323
Model:                            OLS   Adj. R-squared:                  0.323
Method:                 Least Squares   F-statistic:                     1544.
Date:                Wed, 19 Mar 2025   Prob (F-statistic):          1.80e-276
Time:                        05:16:55   Log-Likelihood:                -18574.
No. Observations:                3238   AIC:                         3.715e+04
Df Residuals:                    3236   BIC:                         3.716e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    596.8234     13.226     45.124      0.0

# Interpretation:

Sales=-91.0214*Price+596.8234

Sales at 0 price= 596

For every 1 unit increase in price predicted sales comes down by 91



**Performing Step wise regression- Price V/S Sales,Feature & Display & Evaluating Model Performance With MAE**

In [ ]:
# Fit the linear regression model with multiple independent variables
model = sm.ols('Sales ~ Price + Feature + Display', data=df_train).fit()

# Print the summary of the model
print(model.summary())

# Extract the coefficients
intercept = model.params['Intercept']
price_coeff = model.params['Price']
feature_coeff = model.params['Feature']
display_coeff = model.params['Display']


# Create the 'Predicted Sales' column in df_test
df_test['Predicted Sales'] = intercept + df_test['Price'] * price_coeff + df_test['Feature'] * feature_coeff + df_test['Display'] * display_coeff

# Calculate the Mean Absolute Error (MAE)
mae = (df_test['Sales'] - df_test['Predicted Sales']).abs().mean()

print(f"Mean Absolute Error (MAE): {mae}")


                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.529
Model:                            OLS   Adj. R-squared:                  0.528
Method:                 Least Squares   F-statistic:                     1209.
Date:                Wed, 19 Mar 2025   Prob (F-statistic):               0.00
Time:                        05:16:55   Log-Likelihood:                -17988.
No. Observations:                3238   AIC:                         3.598e+04
Df Residuals:                    3234   BIC:                         3.601e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    323.6978     13.460     24.048      0.0

# Interpretation
Sales=-47*Price+ 63 * Presence Of Add On Feature+ 99 * Presence Of Display

Sales at 0 price= 323

For every 1 unit increase in price predicted sales comes down by 47
Presence of in store display increases sales by 99 Presence of in store add on feature increases sales by 63

**Performing Step wise regression- Price V/S Sales,Feature,Display,Competitor Prices & Evaluating Model Performance With MAE**

In [ ]:
# Fit the linear regression model with all independent variables
independent_vars = '+'.join(df_train.columns.difference(['Sales']))
model = sm.ols(f'Sales ~ {independent_vars}', data=df_train).fit()

# Print the summary of the model
print(model.summary())

# You can then use the model to make predictions on your test data (df_test)
# and evaluate the model's performance.


                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.536
Model:                            OLS   Adj. R-squared:                  0.534
Method:                 Least Squares   F-statistic:                     465.5
Date:                Wed, 19 Mar 2025   Prob (F-statistic):               0.00
Time:                        05:16:55   Log-Likelihood:                -17964.
No. Observations:                3238   AIC:                         3.595e+04
Df Residuals:                    3229   BIC:                         3.600e+04
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept          254.4098     17.818  

# Interpretation:
Sales=-49.4334*Price+ 63.7473 * Feature+ 95.5401 *Display+9.3929* Price Comp_1 +5.9848* Price Comp_2+254.4098

Sales at 0 price= 254

For every 1 unit increase in price predicted sales comes down by 49
For every 1 unit increase in price of competitor1 predicted sales increases by 9.3
For every 1 unit increase in price of competitor4 predicted sales increases by 5.9
Presence of in store display increases sales by 95
Presence of in store add on feature increases sales by 63

From the p vaues, it can be concuded that variables competitor 2,competitor 3 & competitor 5 are insignificant as the p value is greater than 0.05

**Performing Step wise regression- Price V/S Sales,Feature,Display and only the significant Competitor Prices & Evaluating Model Performance With MAE**

In [ ]:
# Fit the linear regression model with selected independent variables
model = sm.ols('Sales ~ Price + Feature + Display + PriceCompetitor1 + PriceCompetitor4', data=df_train).fit()

# Print the summary of the model
print(model.summary())

# You can then use the model to make predictions on your test data (df_test)
# and evaluate the model's performance.

# Example: Predict sales on df_test
df_test['Predicted Sales'] = model.predict(df_test[['Price', 'Feature', 'Display', 'PriceCompetitor1', 'PriceCompetitor4']])

# Calculate the Mean Absolute Error (MAE)
mae = (df_test['Sales'] - df_test['Predicted Sales']).abs().mean()

print(f"Mean Absolute Error (MAE): {mae}")


                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.534
Model:                            OLS   Adj. R-squared:                  0.534
Method:                 Least Squares   F-statistic:                     742.0
Date:                Wed, 19 Mar 2025   Prob (F-statistic):               0.00
Time:                        05:16:55   Log-Likelihood:                -17968.
No. Observations:                3238   AIC:                         3.595e+04
Df Residuals:                    3232   BIC:                         3.598e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept          261.8958     16.926  

# Interpretation:
Sales=-48*Price+ 64 * Feature+ 96 *Display+9.7* Price Comp_1 +6.7* Price Comp_4+261

Sales at 0 price= 261

For every 1 unit increase in price predicted sales comes down by 48
For every 1 unit increase in price of competitor1 predicted sales increases by 9.7
For every 1 unit increase in price of competitor4 predicted sales increases by 6.7
Presence of in store display increases sales by 96
Presence of in store add on feature increases sales by 64

**To study the interaction effect**

In [ ]:
df_train['PriceDisplay_Interaction'] = df_train['Price'] * df_train['Display']


# Understanding Interaction Effects in Marketing Mix Modelling

In real-world marketing scenarios, the impact of one variable often depends on the presence or value of another. This is where interaction effects become important. Rather than modeling variables in isolation, we introduce pairwise interaction terms—products of two variables—to capture these dependencies.

For example, the effect of price on sales might differ depending on whether an in-store display is present. Similarly, the influence of a brand’s price may change depending on the competitor’s pricing, or an advertisement feature may only be effective when combined with a price discount.



**Regression Model With Interaction Effect Of Price & Display(Promotion)**

In [ ]:
# Fit the linear regression model with specified independent variables including interaction term
model = sm.ols('Sales ~ Price + Feature + Display + PriceCompetitor1 + PriceCompetitor4 + PriceDisplay_Interaction', data=df_train).fit()

# Print the summary of the model
print(model.summary())

# Extract the coefficients
intercept = model.params['Intercept']
price_coeff = model.params['Price']
feature_coeff = model.params['Feature']
display_coeff = model.params['Display']
price_competitor1_coeff = model.params['PriceCompetitor1']
price_competitor4_coeff = model.params['PriceCompetitor4']
price_display_interaction_coeff = model.params['PriceDisplay_Interaction']


# Create the 'Predicted Sales' column in df_test
df_test['PriceDisplay_Interaction'] = df_test['Price'] * df_test['Display']
df_test['Predicted Sales'] = (intercept + df_test['Price'] * price_coeff + df_test['Feature'] * feature_coeff +
                              df_test['Display'] * display_coeff + df_test['PriceCompetitor1'] * price_competitor1_coeff +
                              df_test['PriceCompetitor4'] * price_competitor4_coeff +
                              df_test['PriceDisplay_Interaction'] * price_display_interaction_coeff)

# Calculate the Mean Absolute Error (MAE)
mae = (df_test['Sales'] - df_test['Predicted Sales']).abs().mean()

print(f"Mean Absolute Error (MAE): {mae}")


                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.618
Model:                            OLS   Adj. R-squared:                  0.618
Method:                 Least Squares   F-statistic:                     872.7
Date:                Wed, 19 Mar 2025   Prob (F-statistic):               0.00
Time:                        05:16:56   Log-Likelihood:                -17646.
No. Observations:                3238   AIC:                         3.531e+04
Df Residuals:                    3231   BIC:                         3.535e+04
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------
Intercept               

# Achieved the desired results of r^2>0.6